src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -56,7 +55,57 @@ public class DefaultHtmlGenerator : IHtmlGenerator
             IModelMetadataProvider metadataProvider,
             IUrlHelperFactory urlHelperFactory,
             HtmlEncoder htmlEncoder,
-            ClientValidatorCache clientValidatorCache)
+            ClientValidatorCache clientValidatorCache) : this(
+                antiforgery,
+                optionsAccessor,
+                metadataProvider,
+                urlHelperFactory,
+                htmlEncoder,
+                clientValidatorCache,
+                () => new ValidationAttributeProvider(optionsAccessor, metadataProvider, clientValidatorCache))
```

Comment:
:ok:

Yeah we don't need to be concerned about the order of `ArgumentNullException` when you're doing something that will never work